### PR TITLE
QL: Move ParamOrdinal inside TypeResolutions

### PR DESCRIPTION
**Pull request by costin**
Add static imports for enum constants that.

### CODE REVIEW

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/PercentileAggregate.java [hunk 1]
```diff
@@ -92,18 +93,18 @@ protected TypeResolution resolveType() {
...
 
-        resolution = isNumeric(parameter, sourceText(), ParamOrdinal.SECOND);
+        resolution = isNumeric(parameter, sourceText(), SECOND);
         if (resolution.unresolved()) {
             return resolution;
         }
 
-        ParamOrdinal methodOrdinal = ParamOrdinal.fromIndex(parameters().size() + 1);
-        ParamOrdinal methodParameterOrdinal = ParamOrdinal.fromIndex(parameters().size() + 2);
+        ParamOrdinal methodOrdinal = TypeResolutions.ParamOrdinal.fromIndex(parameters().size() + 1);
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/PercentileAggregate.java:106]: same here, `import org.elasticsearch.xpack.ql.expression.TypeResolutions.ParamOrdinal` is already there.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/SqlTypeResolutions.java [hunk 2]
```diff
@@ -18,7 +19,7 @@
 
     private SqlTypeResolutions() {}
 
-    public static TypeResolution isDate(Expression e, String operationName, ParamOrdinal paramOrd) {
+    public static TypeResolution isDate(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/SqlTypeResolutions.java:22]: same here

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/SqlTypeResolutions.java [hunk 3]
```diff
@@ -35,7 +36,7 @@ public static TypeResolution isNumericOrDate(Expression e, String operationName,
             "date", "datetime", "numeric");
     }
 
-    public static TypeResolution isNumericOrDateOrTime(Expression e, String operationName, ParamOrdinal paramOrd) {
+    public static TypeResolution isNumericOrDateOrTime(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/SqlTypeResolutions.java:39]: same here

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/comparison/BinaryComparison.java [hunk 4]
```diff
@@ -33,7 +33,7 @@ public ZoneId zoneId() {
     }
 
     @Override
-    protected TypeResolution resolveInputType(Expression e, Expressions.ParamOrdinal paramOrdinal) {
+    protected TypeResolution resolveInputType(Expression e, TypeResolutions.ParamOrdinal paramOrdinal) {
```
Comment (matriv) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/comparison/BinaryComparison.java:36]: same here

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/arithmetic/ArithmeticOperation.java [hunk 5]
```diff
@@ -25,7 +26,7 @@ protected ArithmeticOperation(Source source, Expression left, Expression right,
     }
 
     @Override
-    protected TypeResolution resolveInputType(Expression e, Expressions.ParamOrdinal paramOrdinal) {
+    protected TypeResolution resolveInputType(Expression e, TypeResolutions.ParamOrdinal paramOrdinal) {
```
Comment (matriv) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/arithmetic/ArithmeticOperation.java:29]: same here

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/logical/BinaryLogic.java [hunk 6]
```diff
@@ -30,7 +31,7 @@ public DataType dataType() {
     }
 
     @Override
-    protected TypeResolution resolveInputType(Expression e, Expressions.ParamOrdinal paramOrdinal) {
+    protected TypeResolution resolveInputType(Expression e, TypeResolutions.ParamOrdinal paramOrdinal) {
```
Comment (matriv) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/logical/BinaryLogic.java:34]: same here

**Review by matriv — APPROVED** [review]
LGTM, minor comments - if it's possible to get rid of the `TypeResolutions` prefix.

**Pull request merged**